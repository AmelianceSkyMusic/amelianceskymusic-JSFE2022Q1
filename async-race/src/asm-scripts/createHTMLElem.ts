export function createHTMLElem(
  parent: HTMLElement | DocumentFragment,
  elementType: string,
  params?: {
    class?: string;
    id?: string;
  },
  text = '',
): HTMLElement {
  const elName$ = document.createElement(elementType);
  if (params?.class) {
    const classes: string[] = params.class.split(' ');
    elName$.classList.add(...classes);
  }
  if (params?.id) {
    elName$.setAttribute('id', params.id);
  }
  if (text) {
    elName$.innerHTML = text;
  }
  parent.append(elName$);

  return elName$;
}
