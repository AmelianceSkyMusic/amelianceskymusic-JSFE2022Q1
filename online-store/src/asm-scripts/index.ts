
export { msg } from './msg';
export { devs } from './devs';
export { createHTMLElem } from './createHTMLElem';
export { shuffleArray } from './shuffleArray';
export { getRandomNumber } from './getRandomNumber';
