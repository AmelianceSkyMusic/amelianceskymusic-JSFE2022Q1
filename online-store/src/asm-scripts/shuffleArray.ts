import { getRandomNumber } from './getRandomNumber';

export function shuffleArray<T>(array: T[]): T[] {
	for (let i = 0; i < array.length; i++) {
		const j = getRandomNumber(0, array.length - 1);
		[array[i], array[j]] = [array[j], array[i]];
	}
	return array;
}
