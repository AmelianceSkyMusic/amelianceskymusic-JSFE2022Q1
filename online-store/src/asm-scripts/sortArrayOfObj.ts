export function sortArrayOfObj<T>(array: T[], key: string, type: string): T[]{
	const arr: T[] = [...array];
	if (key) {
		arr.sort((a: T, b: T): number => {
			const aEl = (a[key as keyof T]);
			const bEl = (b[key as keyof T]);
			if (type === 'num')  {
				if ( +aEl < +bEl){
					return -1;
				}
				if ( +aEl > +bEl){
					return 1;
				}
			} else {
				if ( aEl < bEl){
					return -1;
				}
				if ( aEl > bEl){
					return 1;
				}
			}
			return 0;
		});
	}
	return arr;
}
