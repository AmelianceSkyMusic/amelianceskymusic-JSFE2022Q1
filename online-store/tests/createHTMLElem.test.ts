/**
 * @jest-environment jsdom
 */
import { createHTMLElem } from '../src/asm-scripts/createHTMLElem';

describe('createHTMLElem', () => {
	test('should be render', () => {
		expect(createHTMLElem(document.body, 'div', { class: 'test__class', id: 'test-id'}, 'some test text')).toMatchSnapshot();
	});
});
