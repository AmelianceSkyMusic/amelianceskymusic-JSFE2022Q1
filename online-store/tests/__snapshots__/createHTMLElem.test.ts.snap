// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHTMLElem should be render 1`] = `
<div
  class="test__class"
  id="test-id"
>
  some test text
</div>
`;
